### ISSUE TITLE
2.7.0 provider fails to create new helm_release with values referencing known-after-apply attributes

### ISSUE THREAD

**Issue by jeffreykoetsier** [hashicorp/terraform-provider-helm#965]
### Terraform, Provider, Kubernetes and Helm Versions

```
Terraform version: 1.3.1
Provider version: 2.7.0
Kubernetes version: 1.21.5
```

### Affected Resource(s)
- helm_release

### Terraform Configuration Files
```hcl
terraform { 
  required_providers { 
    helm = { 
      source  = "helm"
      version = "2.7.0"
    } 
  } 
}

provider "helm" { 
  kubernetes {}

  experiments { 
    manifest = true
  } 
}

resource "random_string" "example" {
  length  = 8
  special = false
}

resource "helm_release" "hello" { 
  name      = "hello"
  chart     = "./hello"
  namespace = "default"

  values = [
    <<-EOF
    example: "${random_string.example.result}"
    EOF
  ]
}
```

### Debug Output
https://gist.github.com/jeffreykoetsier/d7e19830b674109b35ac4303dd1b836f

### Steps to Reproduce
1. ensure experiments manifest=true
2. `terraform plan` with an empty state

### Expected Behavior
Propose a plan which creates a new `helm_release` resource.

The above example works as expected with Helm Provider 2.6.0:

```
  # helm_release.hello will be created
  + resource "helm_release" "hello" {
      + atomic                     = false
      + chart                      = "./hello"
      + cleanup_on_fail            = false
      + create_namespace           = false
      + dependency_update          = false
      + disable_crd_hooks          = false
      + disable_openapi_validation = false
      + disable_webhooks           = false
      + force_update               = false
      + id                         = (known after apply)
      + lint                       = false
      + manifest                   = (known after apply)
      + max_history                = 0
      + metadata                   = (known after apply)
      + name                       = "hello"
      + namespace                  = "default"
      + pass_credentials           = false
      + recreate_pods              = false
      + render_subchart_notes      = true
      + replace                    = false
      + reset_values               = false
      + reuse_values               = false
      + skip_crds                  = false
      + status                     = "deployed"
      + timeout                    = 300
      + values                     = (known after apply)
      + verify                     = false
      + version                    = (known after apply)
      + wait                       = true
      + wait_for_jobs              = false
    }
```

### Actual Behavior
The new 2.7.0 provider tries to show the manifest on new release.
If the defined values are referencing attributes on a resource that still has to be created (known after apply), the Helm provider is unable to determine these values. Debugging output shows `{}` for the contents of values.yaml. 

The example Helm chart is validating the passed values by using `required`. As an empty set of values is passed to the dry-run, Helm bails out with an error:

```
│ Error: execution error at (hello/templates/configmap.yaml:6:14): 'example' is empty
```

When using `set { }` syntax instead of the HereDoc notation to define Helm Values, it behaves differently. The Helm provider then magically makes up a value for the "known after apply" `random_string`:

Using `set` instead of HereDoc:
```
  set {
    name  = "example"
    value = random_string.example.result
  }
```

Result:

```
  # helm_release.hello will be created
  + resource "helm_release" "hello" {
    <snip>
      + manifest                   = jsonencode(
            {
              + "configmap/v1/hello" = {
                  + apiVersion = "v1"
                  + data       = {
                      + example = "74D93920-ED26-11E3-AC10-0800200C9A66"
                    }
                  + kind       = "ConfigMap"
                  + metadata   = {
                      + name = "hello"
```

`example` should have the resulting value of `random_string` which generates an 8 character long string. Or actually 'known after apply'. Instead some ID is used.

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by BBBmau** [comment]
Hello! Thank you for opening this issue. We investigated this one. In v2.7.0 we introduced the ability to populate the `manifest` attribute at create time (#903). It looks like we will need to check if any of the values supplied to the resource are unknown, and if so mark `manifest` as computed when that is the case. Because `manifest` is just a string that contains the rendered manifest for the chart and not an object, we are not able to say that specific fields inside it are computed – only the whole `manifest` attribute. This means that resources that rely on unknown values at plan time wont be able to render a value for `manifest` at plan time.
